### ISSUE TITLE
FEEDBACK: Unnecessary friction getting started with the example article

### ISSUE THREAD

**Issue by HenrikBengtsson** [rjournal/rjtools#64]
## TL;DR

The instructions on <https://rjournal.github.io/rjtools/> are incomplete and results in several errors when using a fresh R setup.

I think there's room for improving the instructions, which will lower the friction that a newcomer might experience.  This is especially true if they are new to Rmarkdown, e.g. they've worked their whole life with LaTeX and might even have submitted articles to the R Journal using the LaTeX templates in the past.

## Details

Here is what user who starts out with a fresh R installation, and who doesn't use RStudio, will see when they follow the official
instructions.  We assume the user installs **rjtools** from CRAN using:

```r
$ R --vanilla
R version 4.2.2 (2022-10-31) -- "Innocent and Trusting"
...
> chooseCRANmirror(ind = 1)
> install.packages("rjtools")
...
```

or they may also install from GitHub.

Next, following the 'Getting started with a new article' instructions on <https://rjournal.github.io/rjtools/>, they call:

```r
> library(rjtools)
> create_article()
Error in xfun::with_ext(name, "Rmd") : 
  argument "name" is missing, with no default
```

Hmm ...?!?  It requires to read `?create_article`, or get to the third vignette <https://rjournal.github.io/rjtools/articles/create_article.html>), to get past this. Retrying with:

```r
> library(rjtools)
> create_article("quokka-bilby")
```

That worked.  Continuing on to the 'Your first knit' section.  The instructions there says:

> You can use the regular ways to render the files, either through the RStudio knit button, or using `rmarkdown::render("paper/quokka-bilby.Rmd")` on the R command line.

Let's cut'n'paste that as-is:

```r
> rmarkdown::render("paper/quokka-bilby.Rmd")
Error in abs_path(input) : 
  The file 'paper/quokka-bilby.Rmd' does not exist.
In addition: Warning message:
In normalizePath(x, winslash = winslash, mustWork = must_work) :
  path[1]="paper/quokka-bilby.Rmd": No such file or directory
```

Okay, it could have been that `create_article()` would have created the files in a subfolder `paper/`, but from the error message I guess it's just a generic placeholder.  Lets retry without `paper/`:

```r
> rmarkdown::render("quokka-bilby.Rmd")

processing file: quokka-bilby.Rmd
  |.......                                    |  15% (setup)
  Quitting from lines 35-39 (quokka-bilby.Rmd) 
Error in library(plotly) : there is no package called 'plotly'
```

Righty-oh. At least it's now start processing the Rmd file, but clearly, I'm missing the **plotly** package.  So let's install that:

```r
> chooseCRANmirror(ind = 1)
> install.packages("plotly")
...
```

After installing that, we try again:

```r
> rmarkdown::render("quokka-bilby.Rmd")


processing file: quokka-bilby.Rmd
  |.......                                    |  15% (setup)                   Quitting from lines 35-39 (quokka-bilby.Rmd) 
Error in library(palmerpenguins) : 
  there is no package called 'palmerpenguins'

```

Hmmm... keep on pushing:

```r
> chooseCRANmirror(ind = 1)
> install.packages("palmerpenguins")
...
```

and retrying:

```r
> rmarkdown::render("quokka-bilby.Rmd")

processing file: quokka-bilby.Rmd
output file: quokka-bilby.knit.md

Detected the following packages from article:
  CRAN: ToOoOlTiPs, plotly, crosstalk, tsibbletalk, palmerpenguins, ggplot2
  Bioconductor: 
Error running filter pandoc-citeproc:
Could not find executable pandoc-citeproc
Error: pandoc document conversion failed with error 83
Error: 
! in callr subprocess.
Caused by error in `loadNamespace(x)`:
! there is no package called 'kableExtra'
ℹ See `$stdout` and `$stderr` for standard output and error.
Type .Last.error to see the more details.
```

So another type of error now.  This doesn't look my typical R error. It's complaining about "pandoc".  I wonder if there's an issue with my Pandoc installation (I happens to know what Pandoc is).  Oh wait ... "there is no package called 'kableExtra'".  Let's try by installing that first:

```r
> chooseCRANmirror(ind = 1)
> install.packages("kableExtra")
...
```

Crossing my fingers it'll work this time:

```r
> rmarkdown::render("quokka-bilby.Rmd")

processing file: quokka-bilby.Rmd
output file: quokka-bilby.knit.md

Detected the following packages from article:
  CRAN: ToOoOlTiPs, plotly, crosstalk, tsibbletalk, palmerpenguins, ggplot2
  Bioconductor: 
Error running filter pandoc-citeproc:
Could not find executable pandoc-citeproc
Error: pandoc document conversion failed with error 83
Error: 
! in callr subprocess.
Caused by error: 
! pandoc document conversion failed with error 83
ℹ See `$stdout` and `$stderr` for standard output and error.
```

Uff!  There's still that Pandoc error.

Now we're getting into operating-system specific instructions, which is always tricky.  Never the less, I'm on Ubuntu 22.04, and I've installed Pandoc using:

```sh
$ sudo apt install pandoc
```

However, I was unaware of `pandoc-citeproc`, but trying to call it from the terminal:

```sh
$ pandoc-citeproc
Command 'pandoc-citeproc' not found, but can be installed with:
sudo apt install pandoc-citeproc
```

gives me cut'n'pastable instructions for installing it.  After calling:

```sh
$ sudo apt install pandoc-citeproc
```

and retrying in R:

```r
> rmarkdown::render("quokka-bilby.Rmd")


processing file: quokka-bilby.Rmd
                                                                                                                     
output file: quokka-bilby.knit.md

Detected the following packages from article:
  CRAN: ToOoOlTiPs, plotly, crosstalk, tsibbletalk, palmerpenguins, ggplot2
  Bioconductor: 
/usr/bin/pandoc +RTS -K512m -RTS quokka-bilby.knit.md --to html5 --from markdown+autolink_bare_uris+tex_math_single_backslash --output quokka-bilby.html --lua-filter /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/rmarkdown/rmarkdown/lua/latex-div.lua --lua-filter /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/rjtools/latex-pkg.lua --lua-filter /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/rjtools/sec-depth.lua --self-contained --wrap preserve --standalone --variable toc-float=1 --highlight-style /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/distill/rmarkdown/templates/distill_article/resources/arrow.theme --template /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/distill/rmarkdown/templates/distill_article/resources/default.html '--metadata=link-citations:true' --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34c76da5174html --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34c141ff80fhtml --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34ccf18055html --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34c44bc34c4html --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34c403f80b8html --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34cc38b667html --include-before-body /tmp/henrik/RtmpfkTZpT/file2a34c2b5005a2html --include-before-body /tmp/henrik/RtmpfkTZpT/file2a34c5c3c4223html --include-before-body /tmp/henrik/RtmpfkTZpT/file2a34c1fe88868html --include-after-body /tmp/henrik/RtmpfkTZpT/file2a34c1ab5a891html --include-after-body /tmp/henrik/RtmpfkTZpT/file2a34c3efd7fa5html --include-after-body /tmp/henrik/RtmpfkTZpT/file2a34c581db95bhtml --mathjax --variable 'mathjax-url=https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /tmp/henrik/RtmpfkTZpT/rmarkdown-str2a34c5fa33926.html --include-in-header /tmp/henrik/RtmpfkTZpT/file2a34c758c8ed2html --number-sections --include-in-header /home/henrik/repositories/article-bengtsson-matrixStats/tmp/R_LIBS_USER/rjtools/rjdistill.html --filter /usr/bin/pandoc-citeproc 

Output created: quokka-bilby.html
```

Success!

**Comment by dicook** [comment]
Hi @HenrikBengtsson Thanks for this feedback. Some of this is easy to fix, and we'll make the changes asap.

**Comment by cthombor** [comment]
I'm trying (and so far failing) to figure out how to adjust my RStudio setup so that it'll render `quokka-bilby.Rmd`.  

The offending command is `\RequirePackage{tikz}`.  I can't find `tikz.sty` anywhere in my Win11 filesystem -- it's apparently something that miktex would have in its pgf package.

I had installed RStudio less than a year ago, using tinytex rather than miktex as per the advice of section 1.2  of the [RMarkdown Cookbook](https://bookdown.org/yihui/rmarkdown-cookbook/install-latex.html): "... we recommend that R Markdown users install [TinyTeX.](https://yihui.org/tinytex/)".

I'm getting the impression I'll have more joy with rjtools if I shift to miktex... but then again that could be a hiding to nowhere becuz  in section 1.2 of https://bookdown.org/yihui/rmarkdown-cookbook/install-latex.html: "we recommend that R Markdown users install [TinyTeX.](https://yihui.org/tinytex/)".

I gave tinytex a nudge with a manual invocation of `tinytex::parse_install("RJwrapper.log")`, but it was clueless:

`tlmgr search --file --global "/tikz.sty"`
`Warning message:`
`In parse_packages(...) : Failed to find a package that contains tikz.sty`

My resolution:

1. install miktex, following steps 5 and 6 in the [recipe provided by someone at Reed College](https://www.reed.edu/data-at-reed/software/R/r_studio_pc.html 
2. open the miktex console, click on its "Check for updates" affordance, wait until all updates are complete, then test the installation using step 8 of the [Reed College recipe](https://www.reed.edu/data-at-reed/software/R/r_studio_pc.html)
3. notice that a manual invocation of `tlmgr_search('tikz.sty')` throws an error due to a stale version of Tex Live; 
4. run `tlmgr_update()`; 
5. notice that `tlmgr_search('tikz.sty')` now finds `tikz.sty` in CTAN, hooray!
6. run `rmarkdown::render("quokka-bilby.Rmd")`, and wait *very very* patiently for tinytex to run into fatal error after fatal error, loading one .sty file each time.   

I'd suggest ... that you revise your [Create a Journal Article vignette](https://rjournal.github.io/rjtools/articles/create_article.html) to advise the folks who are using tinytex rather than miktex to perform steps 1, 2, 4 in the list above *before* attempting to do their first rendering of your sample article.   Also I think some warning about being patient is important, as there is no diagnostic output visible in RStudio (at least in my setup) while tinytex is grinding away.  (Viewing the current contents of `RJwrapper.log` in an RStudio edit pane will lock the file and prevent a new logfile from being created when tinytex attempts to recover from its previous fatal error.)

Alternatively... you might work toward providing a template which doesn't rely on style files that are part of miktex but not part of tinytex.   That said... it's nice to have an example with such bells-and-whistles!  

BTW I'm nowhere near wizardly enough in RStudio to figure out how to persuade my installation to use miktex rather than tinytex, otherwise I would have given this a go -- despite the advice in the RMarkdown Cookbook.  I'm *guessing* there's a setup file somewhere on my system that isn't affected by unticking the "Use tinytex when compiling tex files" in Global options for SWeave.   Well ... I'm not going to worry about that, I'll stick with tinytex... now that I have managed to persuade it to compile my first `rjournal_web_article` I can move on to authoring one ;-)

And: thanks for your work on rjtools.  It looks like a nice package!

**Comment by dicook** [comment]
Hi Clark,

Tinytex is the recommendation. The .tex file should be generated even if it doesn’t complete the compile inside RStudio, which would mean that you can latex it outside, if absolutely necessary. On tikz, are you using this to create a figure? It should not be called. What operating system are you working on?

> On 2 Apr 2023, at 4:44 pm, Clark Thomborson ***@***.***> wrote:
> 
> 
> I'm trying (and so far failing) to figure out how to adjust my RStudio setup so that it'll render quokka-bilby.Rmd.
> 
> The offending command is \RequirePackage{tikz}. I can't find tikz.sty anywhere in my Win11 filesystem -- it's apparently something that miktex would have in its pgf package.
> 
> I had installed RStudio less than a year ago, using tinytex rather than miktex as per the advice of section 1.2 of the RMarkdown Cookbook: "... we recommend that R Markdown users install TinyTeX.".
> 
> I'm getting the impression I'll have more joy with rjtools if I shift to miktex... but then again that could be a hiding to nowhere becuz in section 1.2 of https://bookdown.org/yihui/rmarkdown-cookbook/install-latex.html: "we recommend that R Markdown users install TinyTeX.".
> 
> I gave tinytex a nudge with a manual invocation of tinytex::parse_install("RJwrapper.log"), but it was clueless:
> 
> tlmgr search --file --global "/tikz.sty"
> Warning message:
> In parse_packages(...) : Failed to find a package that contains tikz.sty
> 
> My resolution:
> 
> 	• install miktex, following steps 5 and 6 in the [recipe provided by someone at Reed College](https://www.reed.edu/data-at-reed/software/R/r_studio_pc.html
> 	• open the miktex console, click on its "Check for updates" affordance, wait until all updates are complete, then test the installation using step 8 of the Reed College recipe
> 	• notice that a manual invocation of tlmgr_search('tikz.sty') throws an error due to a stale version of Tex Live;
> 	• run tlmgr_update();
> 	• notice that tlmgr_search('tikz.sty') now finds tikz.sty in CTAN, hooray!
> 	• run rmarkdown::render("quokka-bilby.Rmd"), and wait very very patiently for tinytex to run into fatal error after fatal error, loading one .sty file each time.
> I'd suggest ... that you revise your Create a Journal Article vignette to advise the folks who are using tinytex rather than miktex to perform steps 1, 2, 4 in the list above before attempting to do their first rendering of your sample article. Also I think some warning about being patient is important, as there is no diagnostic output visible in RStudio (at least in my setup) while tinytex is grinding away. (Viewing the current contents of RJwrapper.log in an RStudio edit pane will lock the file and prevent a new logfile from being created when tinytex attempts to recover from its previous fatal error.)
> 
> Alternatively... you might work toward providing a template which doesn't rely on style files that are part of miktex but not part of tinytex. That said... it's nice to have an example with such bells-and-whistles!
> 
> BTW I'm nowhere near wizardly enough in RStudio to figure out how to persuade my installation to use miktex rather than tinytex, otherwise I would have given this a go -- despite the advice in the RMarkdown Cookbook. I'm guessing there's a setup file somewhere on my system that isn't affected by unticking the "Use tinytex when compiling tex files" in Global options for SWeave. Well ... I'm not going to worry about that, I'll stick with tinytex... now that I have managed to persuade it to compile my first rjournal_web_article I can move on to authoring one ;-)
> 
> And: thanks for your work on rjtools. It looks like a nice package!
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.
> 

cheers,
Di

---
Dianne Cook
***@***.***

**Comment by cthombor** [comment]
Hi Di, thanks for responding.  This is my very first experience with rjtools.  I was using the example article provided with that package.  This article calls out for rjournal.sty.  The version of `rjournal.sty` that was loaded by my (very recently-installed) version of `rjtools::create_article()` contains a callout to `tikx.sty`.   Perhaps there's a versioning problem somewhere?  IMHO it's quite hazardous to expect all versions of a latex style file to be forward and backward compatible.... suggesting to me that ... it might save a lot of time (for both users and rjtool maintainers) if all style files required to compile the quokka-bilby.Rmd were loaded into the project directory by `rjtools::create_article()`

I'm glad to know that you're recommending tinytex for rjtools.  The (embedded) callout to tikx.sty made me think that there might be some other miktex dependencies... and I'm now thinking that there's probably no need to install miktex at all, given that tinytex is so diligent about digging through CTAN to find style files.

BTW latex has bugger-all in the way of support for version control, so variant versions of style files -- and the many versions of latex itself -- can cause great difficulties when compiling someone else's latex, or for that matter when compiling latex you  authored yourself at some future date, unless you're careful to keep copies of *all* of the style files you used when compiling.  I note that R Journal expects authors to include their style files with a submission.   Possibly the dependency on tikx.sty is a weak point in its production processes?

**Comment by dicook** [comment]
I think you mean tikz.sty. This would be because some legacy latex papers use tikz for figures. When the style was purely latex there were some very strict rules on what an author could do, for the reasons you describe, it was a nightmare for us to compile the papers for each issue. 

You didn’t tell me what operating system you are using. It might help to provide the sessionInfo also.

> On 3 Apr 2023, at 8:00 am, Clark Thomborson ***@***.***> wrote:
> 
> 
> Hi Di, thanks for responding. This is my very first experience with rjtools. I was using the example article provided with that package. This article calls out for rjournal.sty. The version of rjournal.sty that was loaded by my (very recently-installed) version of rjtools::create_article() contains a callout to tikx.sty. Perhaps there's a versioning problem somewhere? IMHO it's quite hazardous to expect all versions of a latex style file to be forward and backward compatible.... suggesting to me that ... it might save a lot of time (for both users and rjtool maintainers) if all style files required to compile the quokka-bilby.Rmd were loaded into the project directory by rjtools::create_article()
> 
> I'm glad to know that you're recommending tinytex for rjtools. The (embedded) callout to tikx.sty made me think that there might be some other miktex dependencies... and I'm now thinking that there's probably no need to install miktex at all, given that tinytex is so diligent about digging through CTAN to find style files.
> 
> BTW latex has bugger-all in the way of support for version control, so variant versions of style files -- and the many versions of latex itself -- can cause great difficulties when compiling someone else's latex, or for that matter when compiling latex you authored yourself at some future date, unless you're careful to keep copies of all of the style files you used when compiling. I note that R Journal expects authors to include their style files with a submission. Possibly the dependency on tikx.sty is a weak point in its production processes?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.
> 

cheers,
Di

---
Dianne Cook
***@***.***

**Comment by cthombor** [comment]
Oh you're right, it's tikz.sty.  The callout in RJournal.sty is `\RequirePackage{tikz}`.  Apologies.

https://tex.stackexchange.com/questions/9518/how-to-specify-latex-style-file-precedence-when-multiple-versions-might-exist suggests that the working directory is searched first for .sty files, in a typical LaTeX installation.  That's my experience as well.  


But!  Tikz is a very special package in miktex, according to https://comp.text.tex.narkive.com/TEjcuqxT/sty-files-in-subdirectory it "... uses its own interpreter for libraries".   Apparently this interpreter doesn't look in the "/sty" subdirectory of its working directory... which is where I usually put the style files when collaborating with co-authors, back in the days when I was actively doing this... 

All to say that I'd expect style-file versioning to create some ongoing "fun" for rjtool users and devs!

**Comment by cthombor** [comment]
Here's my OS and session info.

Edition	Windows 11 Pro
Version	22H2
Installed on	‎21/‎09/‎2022
OS build	22621.1413
Serial number	YX03F6BD
Experience	Windows Feature Experience Pack 1000.22639.1000.0


> sessioninfo::session_info()
─ Session info ────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.3 (2023-03-15 ucrt)
 os       Windows 10 x64 (build 22621)
 system   x86_64, mingw32
 ui       RStudio
 language (EN)
 collate  English_New Zealand.utf8
 ctype    English_New Zealand.utf8
 tz       Pacific/Auckland
 date     2023-04-03
 rstudio  2023.03.0+386 Cherry Blossom (desktop)
 pandoc   2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ────────────────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 cli           3.6.1   2023-03-23 [1] CRAN (R 4.2.3)
 colorspace    2.1-0   2023-01-23 [1] CRAN (R 4.2.3)
 digest        0.6.31  2022-12-11 [1] CRAN (R 4.2.2)
 evaluate      0.20    2023-01-17 [1] CRAN (R 4.2.3)
 fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.2.3)
 glue          1.6.2   2022-02-24 [1] CRAN (R 4.2.1)
 htmltools     0.5.5   2023-03-23 [1] CRAN (R 4.2.3)
 httr          1.4.5   2023-02-24 [1] CRAN (R 4.2.3)
 kableExtra    1.3.4   2021-02-20 [1] CRAN (R 4.2.3)
 knitr         1.42    2023-01-25 [1] CRAN (R 4.2.3)
 lifecycle     1.0.3   2022-10-07 [1] CRAN (R 4.2.1)
 magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.2.1)
 munsell       0.5.0   2018-06-12 [1] CRAN (R 4.2.1)
 R6            2.5.1   2021-08-19 [1] CRAN (R 4.2.1)
 rlang         1.1.0   2023-03-14 [1] CRAN (R 4.2.3)
 rmarkdown     2.21    2023-03-26 [1] CRAN (R 4.2.3)
 rstudioapi    0.14    2022-08-22 [1] CRAN (R 4.2.1)
 rvest         1.0.3   2022-08-19 [1] CRAN (R 4.2.2)
 scales        1.2.1   2022-08-20 [1] CRAN (R 4.2.1)
 sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.2.1)
 stringi       1.7.12  2023-01-11 [1] CRAN (R 4.2.2)
 stringr       1.5.0   2022-12-02 [1] CRAN (R 4.2.2)
 svglite       2.1.1   2023-01-10 [1] CRAN (R 4.2.3)
 systemfonts   1.0.4   2022-02-11 [1] CRAN (R 4.2.1)
 viridisLite   0.4.1   2022-08-22 [1] CRAN (R 4.2.1)
 webshot       0.5.4   2022-09-26 [1] CRAN (R 4.2.3)
 xfun          0.38    2023-03-24 [1] CRAN (R 4.2.3)
 xml2          1.3.3   2021-11-30 [1] CRAN (R 4.2.1)
 yaml          2.3.7   2023-01-23 [1] CRAN (R 4.2.3)

 [1] C:/Users/cthom/AppData/Local/R/win-library/4.2
 [2] C:/Program Files/R/R-4.2.3/library

───────────────────────────────────────────────────────────────────────────────────────

**Comment by dicook** [comment]
Thanks!

> On 3 Apr 2023, at 8:21 am, Clark Thomborson ***@***.***> wrote:
> 
> 
> Here's my OS and session info.
> 
> Edition Windows 11 Pro
> Version 22H2
> Installed on ‎21/‎09/‎2022
> OS build 22621.1413
> Serial number YX03F6BD
> Experience Windows Feature Experience Pack 1000.22639.1000.0
> 
> sessioninfo::session_info()
> ─ Session info ────────────────────────────────────────────────────────────────────────
> setting value
> version R version 4.2.3 (2023-03-15 ucrt)
> os Windows 10 x64 (build 22621)
> system x86_64, mingw32
> ui RStudio
> language (EN)
> collate English_New Zealand.utf8
> ctype English_New Zealand.utf8
> tz Pacific/Auckland
> date 2023-04-03
> rstudio 2023.03.0+386 Cherry Blossom (desktop)
> pandoc 2.19.2 @ C:/Program Files/RStudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
> 
> ─ Packages ────────────────────────────────────────────────────────────────────────────
> package * version date (UTC) lib source
> cli 3.6.1 2023-03-23 [1] CRAN (R 4.2.3)
> colorspace 2.1-0 2023-01-23 [1] CRAN (R 4.2.3)
> digest 0.6.31 2022-12-11 [1] CRAN (R 4.2.2)
> evaluate 0.20 2023-01-17 [1] CRAN (R 4.2.3)
> fastmap 1.1.1 2023-02-24 [1] CRAN (R 4.2.3)
> glue 1.6.2 2022-02-24 [1] CRAN (R 4.2.1)
> htmltools 0.5.5 2023-03-23 [1] CRAN (R 4.2.3)
> httr 1.4.5 2023-02-24 [1] CRAN (R 4.2.3)
> kableExtra 1.3.4 2021-02-20 [1] CRAN (R 4.2.3)
> knitr 1.42 2023-01-25 [1] CRAN (R 4.2.3)
> lifecycle 1.0.3 2022-10-07 [1] CRAN (R 4.2.1)
> magrittr 2.0.3 2022-03-30 [1] CRAN (R 4.2.1)
> munsell 0.5.0 2018-06-12 [1] CRAN (R 4.2.1)
> R6 2.5.1 2021-08-19 [1] CRAN (R 4.2.1)
> rlang 1.1.0 2023-03-14 [1] CRAN (R 4.2.3)
> rmarkdown 2.21 2023-03-26 [1] CRAN (R 4.2.3)
> rstudioapi 0.14 2022-08-22 [1] CRAN (R 4.2.1)
> rvest 1.0.3 2022-08-19 [1] CRAN (R 4.2.2)
> scales 1.2.1 2022-08-20 [1] CRAN (R 4.2.1)
> sessioninfo 1.2.2 2021-12-06 [1] CRAN (R 4.2.1)
> stringi 1.7.12 2023-01-11 [1] CRAN (R 4.2.2)
> stringr 1.5.0 2022-12-02 [1] CRAN (R 4.2.2)
> svglite 2.1.1 2023-01-10 [1] CRAN (R 4.2.3)
> systemfonts 1.0.4 2022-02-11 [1] CRAN (R 4.2.1)
> viridisLite 0.4.1 2022-08-22 [1] CRAN (R 4.2.1)
> webshot 0.5.4 2022-09-26 [1] CRAN (R 4.2.3)
> xfun 0.38 2023-03-24 [1] CRAN (R 4.2.3)
> xml2 1.3.3 2021-11-30 [1] CRAN (R 4.2.1)
> yaml 2.3.7 2023-01-23 [1] CRAN (R 4.2.3)
> 
> [1] C:/Users/cthom/AppData/Local/R/win-library/4.2
> [2] C:/Program Files/R/R-4.2.3/library
> 
> ───────────────────────────────────────────────────────────────────────────────────────
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.
> 

cheers,
Di

---
Dianne Cook
***@***.***

**Comment by dicook** [comment]
@cthombor Can you tell me if the html file is created? If so, this will be sufficient for a submission. 

I just tested with someone on Windows, and they had no problems. So I suspect it is related to your tex installation. It would be good to know if a .tex file is created on your system.

**Comment by cthombor** [comment]
Yes I did manage to create a .tex file.  The pdf looks good.  

The problem was indeed with my tex installation, which is why I suggested you adjust your vignette to inform people that they'll get no joy from rjtools if their tex installation is stale.  (That was step 4 in my defect-resolution.)   

I'm now almost certain that a miktex installation isn't required (that was my steps 1 & 2).

I still think it'd be a good idea to adjust your vignette so that it warns people that their first compilation of quokka-bilby.Rmd may take a long time, and there'll be no console output while tinytex is hitting fatal error after fatal error (making a step of forward progress each time by loading just one of the required .sty files).  

I also think it'd be a good idea to adjust your template so that it is self-contained i.e. so that it is *not* reliant on importing .sty files from CTAN (not only because that's slow, but also because there's always a risk that some later version of these .sty files will *not* be backward-compatible with your template).   Alternatively I suppose you could warn rjtool users of the possibility that they *might* encounter a version-incompatibility, how they could recognise this (e.g. by diffing the pdf they generate with some reference version), and what they should do if this happens.

**Comment by dicook** [comment]
Ok, noted.

**Comment by HenrikBengtsson** [comment]
I still get:

```r
Quitting from lines 35-40 [setup] (test.Rmd)
Error in library(kableExtra) : there is no package called 'kableExtra'
```

after using `rjtools::create_article()`.  This is with:

```r
> packageVersion("rjtools")
[1] '1.0.12'
```
